internal/controllers/user_integrations_controller.go

```diff
@@ -1810,6 +1812,20 @@ func (udc *UserDevicesController) registerDeviceTesla(c *fiber.Ctx, logger *zero
 		return fiber.NewError(fiber.StatusBadRequest, "Couldn't parse request body.")
 	}
 
+	if reqBody.ExternalID == "" {
+		return fiber.NewError(fiber.StatusBadRequest, "Missing externalID parameter")
+	}
+
+	if apiVersion == 2 { // If version is 2, we are using fleet api which, which has token stored in cache
+		deviceIntReq, err := udc.getTeslaAuthFromCache(c.Context(), ud.UserID)
+		if err != nil {
+			return fiber.NewError(fiber.StatusBadRequest, "Couldn't parse request body.")
```

Comment:
It's not quite the case that we couldn't parse the request body, is it? We definitely parsed the request body. "Couldn't retrieve stored credentials."?